pycmds/hardware/filters.py

```diff
@@ -16,45 +20,91 @@
+        self.yaqd_host = kwargs.get("yaqd_host", "127.0.0.1")
+        self.motor = yaqc.Client(self.yaqd_port, host=self.yaqd_host)
+        self.motor_units = self.motor.get_units()
+        if self.motor_units == "deg":
+            self.motor_units = "deg"
+        self.native_units = kwargs.get("native_units", "deg")
+        self.native_per_motor = float(wt.units.convert(1, self.motor_units, self.native_units))
+        hw.Driver.__init__(self, *args, **kwargs)
+        id_ = self.motor.id()
+        if id_["model"] is not None:
+            self.hardware.model = id_["model"]
+        elif id_["kind"].startswith("fake"):
+            self.hardware.model = "fake"
+        else:
+            self.hardware.model = id_["kind"]
+
         self.factor = self.hardware.factor
         self.factor.write(kwargs["factor"])
         self.motor_limits = self.hardware.motor_limits
+        self.motor_limits.write(*self.motor.get_limits())
         self.motor_position = self.hardware.motor_position
         self.zero_position = self.hardware.zero_position
-        self.update_recorded()
-        self.native_per_deg = 1
-        hw.Driver.__init__(self, *args, **kwargs)
+        self.set_zero(self.zero_position.read(self.motor_units))
+        self.recorded["_".join([self.name, "zero"])] = [
+            self.zero_position,
+            "deg",
+            0,
+            self.label.read(),
+            True,
+        ]
+        
+    def initialize(self):
+        # This should be unnecessary at some point, once everything is yaq
+        self.get_position()
+        self.initialized.write(True)
+        self.initialized_signal.emit()
+        
+    def is_busy(self):
+        return self.motor.busy()
+        
+    def get_position(self):
+        position = self.motor.get_position()
+        self.motor_position.write(position)
+        filter = (
+            (position - self.zero_position.read(self.motor_units))
+            * self.native_per_motor
+            * self.factor.read()
+        )
+        self.position.write(filter, self.native_units)
+        return filter
 
     def get_state(self):
         state = super().get_state()
-        state["zero_position"] = self.zero_position.read(self.motor_units)
-        state["motor_position"] = self.motor_position.read(self.motor_units)
-
+        state["zero_position"] = float(self.zero_position.read(self.motor_units))
+        return state
+    
     def load_state(self, state):
         super().load_state(state)
-        self.motor_position.write(state["motor_position"])
-        self.zero_position.write(state["zero_position"])
-
+        self.hardware.zero_position.write(state.get("zero_position", 0))
+        
     def set_motor_position(self, motor_position):
-        self.motor_position.write(motor_position)
-        self.save_status()
-
+        self.motor.set_position(motor_position)
+        time.sleep(0.01)
+        while self.is_busy():
+            self.get_position()
```

Comment:
```suggestion
            self.get_position()
            time.sleep(0.001)
```

Reply:
Actually.... The whole while loop should just be replaced with `self.wait_until_still()`

Reply:
Does `wait_until_still()` try to minimize wait time while still keeping overhead down?  I cannot say what the ideal wait time is for our types of scans when we like to minimize the delay between moving equipment and acquiring data,  ignoring slow devices/hardware that may need ms.wait()

Reply:
wait until still I think does 10ms jumps... plenty fine, in my opinion

ms_wait is a separately controlled thing, don't worry about it right now.

This is hardware anyway, ms_wait only applies to sensors.

Reply:
Wait until still does nothing to change the actual motion, only the polling frequency that we check to see that it is done.

Reply:
I mean, we always like to put as many of our photons to work as possible, to keep the time spent moving equipment as low as we can.  The amount of effort we put into doing so, and the possibility of going *too* fast, is a problem we often vet.